### PR TITLE
TORPEDOS: variational pose estimation

### PR DESCRIPTION
**Pull request by DSsoto**
Got a variational pose estimation approach working consistently, using SciPy, at least with current bagged data. Fixed segfaults due to lazy multi-threading.

@jpanikulam Doesn't look like there's any funky git stuff happening this time but double check just in case.

### CODE REVIEW

File: gnc/sub8_perception/srv/TorpBoardPoseRequest.srv [hunk 1]
```diff
@@ -1,6 +1,6 @@
 geometry_msgs/PoseStamped pose_stamped
-int32[12] l_proj_mat
-int32[12] r_proj_mat
+float32[12] l_proj_mat
+float32[12] r_proj_mat
```
Comment (jpanikulam) [inline gnc/sub8_perception/srv/TorpBoardPoseRequest.srv:3]: Ah

File: gnc/sub8_perception/CMakeLists.txt [hunk 2]
```diff
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 2.8.3)
 project(sub8_perception)
 
 SET(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake;${CMAKE_MODULE_PATH}")
-SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -g -rdynamic -Wall -std=c++11")
+SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fopenmp -g -rdynamic -Wall")
```
Comment (jpanikulam) [inline gnc/sub8_perception/CMakeLists.txt:5]: The now up-to-date CMakeLists in sub8_perception seems to fix the issues with inability to compile with C++11

File: gnc/sub8_perception/srv/TorpBoardPoseRequest.srv (hunk 1 above)
Comment (DSsoto) [inline gnc/sub8_perception/srv/TorpBoardPoseRequest.srv:3]: lol I don't know how I didn't catch it when you made it so explicit last time...

File: gnc/sub8_perception/CMakeLists.txt (hunk 2 above)
Comment (jpanikulam) [inline gnc/sub8_perception/CMakeLists.txt:5]: incase you want to give it a go

Comment (DSsoto) [inline gnc/sub8_perception/CMakeLists.txt:5]: I'll have that in mind moving forward. BTW, whatever you did drastically sped up compilation times, thank you for that.

Comment (jpanikulam) [inline gnc/sub8_perception/CMakeLists.txt:5]: Hahahahah! I have some more tricks up my sleeve (with some thanks to Forrest) to further speed up!

**Pull request merged**